mesh/generator/generator.go

```diff
@@ -185,6 +185,65 @@ func BuildMeshMap(ctx context.Context, o mesh.Options, gi *mesh.GlobalInfo) (mes
 			mesh.CheckError(err)
 		}
 
+		// if ambient, add ztunnel
+		if gi.KialiCache.IsAmbientEnabled(cp.Cluster.Name) {
+			ztunnels, err := gi.Business.Workload.GetAllWorkloads(ctx, cp.Cluster.Name, fmt.Sprintf("%s=%s", config.IstioAppLabel, config.Ztunnel))
+			mesh.CheckError(err)
+
+			for _, ztunnel := range ztunnels {
+				var infraData interface{}
+
+				if len(ztunnel.Pods) > 0 {
+					dump := gi.Business.Workload.GetZtunnelConfig(ztunnel.Cluster, ztunnel.Namespace, ztunnel.Pods[0].Name)
```

Comment:
There is also another cached method for the dump: 
`dump := gi.KialiCache.GetZtunnelDump(ztunnel.Cluster, ztunnel.Namespace, ztunnel.Pods[0].Name)`

Reply:
This call is a wrapper for the cached call, so it should be OK.